src/cryptonote_core/blockchain.cpp

```diff
@@ -1056,17 +1068,17 @@ bool Blockchain::switch_to_alternative_blockchain(std::list<block_extended_info>
 
   // pop blocks from the blockchain until the top block is the parent
   // of the front block of the alt chain.
-  std::list<block> disconnected_chain;
+  std::list<block_and_checkpoint> disconnected_chain; // TODO(loki): use a vector and rbegin(), rend() because we don't have push_front
   while (m_db->top_block_hash() != alt_chain.front().bl.prev_id)
   {
-    block b = pop_block_from_blockchain();
-    disconnected_chain.push_front(b);
+    block_and_checkpoint entry = {};
+    entry.block                = pop_block_from_blockchain();
+    entry.checkpointed         = m_db->get_block_checkpoint(cryptonote::get_block_height(entry.block), entry.checkpoint);
+    disconnected_chain.push_front(entry);
   }
 
   auto split_height = m_db->height();
-
-  // TODO(loki): This is a work around for sometimes reorganising the blockchain causing inconsistent difficulty values
-  LOKI_DEFER
+  LOKI_DEFER // TODO(loki): This is a work around for sometimes reorganising the blockchain causing inconsistent difficulty values
   {
     if (nettype() == MAINNET)
```

Comment:
Why did we only do this diff fixup on mainnet?

Reply:
So we can leave testnet running and see if we stop getting difficulty problems having inadvertently fixed it somehow.

Reply:
Oh, right.